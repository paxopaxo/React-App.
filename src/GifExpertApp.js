import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GrifGrid } from './components/GrifGrid'

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories )
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
            <ol>
                {
                    categories.map( category => ( // () retorna implicitamente
                        <GrifGrid
                            category={category} 
                            key={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}
