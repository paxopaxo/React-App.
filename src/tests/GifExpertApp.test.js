import React from 'react'


import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"
import { expressionStatement } from '@babel/types'

describe('Pruebas en GifExpertApp ', () => {

    test('Debe mostrarse correctamente', () => {
        
        const wrapper = shallow( <GifExpertApp/> )

        expect(wrapper).toMatchSnapshot()
    })
    
    test('debe mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/>)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GrifGrid').length ).toBe( categories.length )

    })
    

})
