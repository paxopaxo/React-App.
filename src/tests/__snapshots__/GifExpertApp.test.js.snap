// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp  Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp  debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GrifGrid
      category="One Punch"
      key="One Punch"
    />
    <GrifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
